R/pmcmc.R

```diff
@@ -61,13 +66,32 @@
 ##'
 ##' @export
 pmcmc <- function(pars, filter, n_steps, save_state = TRUE,
-                  save_trajectories = FALSE, progress = FALSE) {
+                  save_trajectories = FALSE, progress = FALSE,
+                  n_chains = 1) {
   assert_is(pars, "pmcmc_parameters")
   assert_is(filter, "particle_filter")
   assert_scalar_positive_integer(n_steps)
   assert_scalar_logical(save_state)
   assert_scalar_logical(save_trajectories)
+  assert_scalar_positive_integer(n_chains)
+
+  if (n_chains == 1) {
+    pmcmc1(pars, filter, n_steps, save_state, save_trajectories, progress)
+  } else {
+    samples <- vector("list", n_chains)
+    for (i in seq_along(samples)) {
+      if (progress) {
+        message(sprintf("Running chain %d / %d", i, n_chains))
+      }
+      samples[[i]] <-
+        pmcmc1(pars, filter, n_steps, save_state, save_trajectories, progress)
+    }
+    pmcmc_combine(samples = samples)
+  }
+}
 
+pmcmc1 <- function(pars, filter, n_steps, save_state, save_trajectories,
```

Comment:
`pmcmc1` for some reason doesn't sit right with me. `pmcmc_single_chain` or `pmcmc_chain` feel more descriptive. Definitely very subjective though, so feel free to keep if you prefer the current name

Reply:
sure, good call. It's an internal function but I do not feel strongly at all. pmcmc_single_chain is good